framework/include/samplers/Sampler.h

```diff
@@ -18,168 +18,149 @@
+  dof_id_type getNumberOfCols() const;
+  dof_id_type getNumberOfLocalRows() const;
 
+  ///@{
   /**
-   * Store the state of the MooseRandom generator so that new calls to getSamples will create
-   * new numbers.
+   * Return the beginning/end local row index for this processor
    */
-  void execute();
+  dof_id_type getLocalRowBegin() const;
+  dof_id_type getLocalRowEnd() const;
+  ///@}
 
+protected:
+  ///@{
   /**
-   * Return the Sample::Location for the given multi app index.
-   * @param global_index The global row, which is the row if all the DenseMatrix objects were
-   * stacked in order in a single object. When using SamplerTransientMultiApp or
-   * SamplerFullSolveMultiApp the global_index is the MultiApp global index.
-   * @return The location which includes the DenseMatrix index and the row within that matrix.
+   * These methods should be called within the constructor of child classes to define the size of
+   * the matrix to be created.
    */
-  Sampler::Location getLocation(dof_id_type global_index);
+  void setNumberOfRows(dof_id_type n_rows);
+  void setNumberOfCols(dof_id_type n_cols);
+  ///@}
 
   /**
-   * Return the number of samples.
-   * @return The total number of rows that exist in all DenseMatrix values from getSamples()
+   * Set the number of seeds required by the sampler. The Sampler will generate
+   * additional seeds as needed. This function should be called in the constructor
+   * of child objects.
+   * @param number The required number of random seeds, by default this is called with 1.
    */
-  dof_id_type getTotalNumberOfRows();
+  void setNumberOfRandomSeeds(std::size_t number);
 
   /**
-   * Return the number of rows local to this processor.
+   * Get the next random number from the generator.
+   * @param index The index of the seed, by default this is zero. To add additional seeds
+   *              indices call the setNumberOfRequiedRandomSeeds method.
+   *
+   * @return A double for the random number, this is double because MooseRandom class uses double.
    */
-  dof_id_type getLocalNumerOfRows();
+  double rand(unsigned int index = 0);
 
   /**
-   * Return the beginning local row index for this processor
+   * Base class must override this method to supply the sample distribution data.
+   * @param row_index The row index of sample value to compute.
+   * @param col_index The column index of sample value to compute.
+   * @return The value for the given row and column.
    */
-  dof_id_type getLocalRowBegin();
+  virtual Real computeSample(dof_id_type row_index, dof_id_type col_index) = 0;
 
+  ///@{
   /**
-   * Return the ending local row index for this processor
+   * Setup method called prior and after looping through distributions.
+   *
+   * These methods should not be called directly, each is automatically called by the public
+   * getSamples() or getLocalSamples() methods.
    */
-  dof_id_type getLocalRowEnd();
+  virtual void sampleSetUp(){};
```

Comment:
Could you add comments for these two functions?

Reply:
The comments exist, the diff just doesn't show them.